src/lib/mina_lib/mina_lib.ml

```diff
@@ -2157,6 +2157,14 @@ let create ~commit_id ?wallets (config : Config.t) =
             }
           in
 
+          let ledger_backing =
```

Comment:
Maybe this could be an exposed top-level

```
let ledger_backing_type t = match t.hardfork_mode with
 ...
```

definition and re-used in `mina_cli_entrypoint.ml`? Or the backing type can be stored in the `Mina_lib.Config.t` type so it doesn't have to be recomputed.

Reply:
backing type is inferred from config, so I'd like to avoid so. 

It seems we don't have a explicit separation between config and state. I think with a potential configuration refactor in the future paying off the debt would need to explicitly seperate the 2.